### PR TITLE
[DAGISel][ARM] Fix vector truncate combine for big-endian

### PR DESCRIPTION
**Pull request by ostannard**
This DAG combine was incorrect for big-endian targets, because it assumes that when a bitcast changes the lane width, the least-significant bits of the wider lanes are in the lower-numbered lanes of the smaller type, which is only true for little-endian.

### CODE REVIEW

File: llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll [hunk 1]
```diff
@@ -0,0 +1,31 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=armebv7-unknown-none-eabihf -mattr=+neon < %s | FileCheck %s
+
+define i32 @test(i64 %arg1) "target-features"="+neon" {
```
[the remark this answers is not in the document]
Comment (ostannard) [reply llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll]: Done

File: llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll [hunk 2]
```diff
@@ -0,0 +1,31 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=armebv7-unknown-none-eabihf -mattr=+neon < %s | FileCheck %s
```
Comment (RKSimon) [inline llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll:2]: is there any existing ARM codegen test file that does something similar that we're just missing BE test coverage?

Comment (ostannard) [reply llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll:2]: I've had a search and can't find any existing tests which cover this. The only way I've been able to hit this code path is with operations on `<8 x i64>`, which is very over-sized for NEON so gets expanded multiple times.

Comment (RKSimon) [reply llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll:2]: OK - please rename this vector-trunc.ll and add equivalent LE test checks as well.

Comment (ostannard) [reply llvm/test/CodeGen/ARM/big-endian-vector-trunc.ll]: Done

**Review by arsenm — APPROVED** [review]

**Review by RKSimon — APPROVED** [review]
LGTM with one minor

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**